core/src/main/java/com/backblaze/b2/json/B2JsonObjectHandler.java

```diff
@@ -162,21 +159,38 @@ protected void initializeImplementation(B2JsonHandlerMap handlerMap) throws B2Js
 
         // Figure out the argument positions for the constructor.
         {
+            // Parse @B2Json.constructor#params into an array
             String paramsWithCommas = annotation.params().replace(" ", "");
-            String [] paramNames = paramsWithCommas.split(",");
-            if (paramNames.length == 1 && paramNames[0].length() == 0) {
-                paramNames = new String [0];
+            String[] annotationParamNames = paramsWithCommas.split(",");
+            if (annotationParamNames.length == 1 && annotationParamNames[0].length() == 0) {
+                annotationParamNames = null;
             }
 
-            final int constructorParamCount = fields.length + numberOfVersionParams;
-            if (paramNames.length != constructorParamCount) {
+            // Verify that, if present, the number of params specified in the annotation is correct
+            final int expectedParamCount = fields.length + numberOfVersionParams;
+            if (annotationParamNames != null && annotationParamNames.length != expectedParamCount) {
+                throw new IllegalArgumentException(clazz.getName() + " constructor's @B2Json.constructor annotation does not have the right number of params.");
+            }
+
+            // Verify that the number of actual constructor params is correct
+            Parameter[] constructorParams = this.constructor.getParameters();
```

Comment:
Does this work even if javac is not called with the `-parameters` argument?

Reply:
Yes. If that option is removed from  `b2sdk.gradle.kts`, all tests pass except those in `B2JsonInferredParametersTest`, which fail with "[class] constructor is missing 'params' for its @B2Json.constructor annotation. Either specify this or add -parameters to javac args.

When `-parameters` is not enabled, the only differences that affect our usage are: 
* `Parameter#isNamePresent()` returns false
* `Parameter#getName()` returns `arg0`, `arg1`, etc.